src/Build/Evaluation/Evaluator.cs

```diff
@@ -2367,12 +2378,12 @@ private void SetAllProjectsProperty()
             if (_lastModifiedProject != null)
             {
                 P oldValue = _data.GetProperty(Constants.MSBuildAllProjectsPropertyName);
-
+                string streamImports = string.Join(";", _streamImports.ToArray());
                 P newValue = _data.SetProperty(
                     Constants.MSBuildAllProjectsPropertyName,
                     oldValue == null
-                        ? _lastModifiedProject.FullPath
-                        : $"{_lastModifiedProject.FullPath};{oldValue.EvaluatedValue}",
+                        ? $"{_lastModifiedProject.FullPath};{streamImports}"
+                        : $"{_lastModifiedProject.FullPath};{streamImports};{oldValue.EvaluatedValue}",
```

Comment:
Hmm. I don't love having extra semicolons in this list, which we will now in the "normal" (no stream save) case. But not a huge deal, everything should be perfectly capable of stripping them out later.